### DISCUSSION TITLE
Reactive body attributes using useMeta

### DISCUSSION THREAD

**Discussion by michaeljolley** [nuxt/nuxt#18121] [Questions] [answered]
The [documentation for meta tags](https://v3.nuxtjs.org/docs/usage/meta-tags) suggest modifying body attributes with `useMeta` as shown below:

```js
export default {
  setup () {
    useMeta({
      meta: [
        { name: 'viewport', content: 'width=device-width, initial-scale=1, maximum-scale=1' }
      ],
      bodyAttrs: {
        class: 'test'
      }
    }
  }
}
```
In my instance, I'm attempting to add a class based on a composable, but the class is never set even though I see the value is being reactively updated. 

I'm sure I'm missing something obvious. My `app.vue` is:

```js
<script setup>
  const { dark } = useDark();
  useMeta({
    bodyAttrs: {
      class: dark ? 'dark' : '',
    },
  });
</script>
<template>
  <div>
    App: {{ dark }}
    <NuxtPage />
  </div>
</template>
<style>
.dark {
  background-color: red;
  color: #fff;
}
</style>
```

I've created a [StackBlitz reproducing the issue](https://stackblitz.com/edit/github-86uulq?file=app.vue).

**Comment by viii47** [comment, marked as answer]
Hey @MichaelJolley,

I think you need to signal the `useMeta()` Method that the class attribute is reactive.

```js
    useMeta({
      bodyAttrs: {
        class: computed(() => dark ? 'dark' : '')
      }
    })
```

Source: [@vueuse/head](https://github.com/vueuse/head)

**Reply by michaeljolley** [reply]
Thanks a ton. That certainly pointed me in the right direction. Final version was:

```js
useMeta({
  bodyAttrs: {
    class: computed(() => dark.value ? 'dark' : '')
  }
})
```

**Reply by pratyushtewari** [reply]
YES!! this is what I was missing.

I needed to set the class on body while using the `@nuxtjs/color-mode`

so I added this to the app.vue

```
<script setup lang="ts">
const colorMode = useColorMode();
useHead({
  bodyAttrs : {
    class : computed(() => colorMode.preference)
  },
});
```

and now I can easily see the call added to the body tag, saved in localstorge and updated in realtime.

Thanks!
